models/ldndc/R/met2model.LDNDC.R

```diff
@@ -0,0 +1,247 @@
+  
+  
+  # Check the nc-files
+  nc_file <- list.files(in.path)[grep("*.nc", list.files(in.path))]
+  
+  # Set-up the outfile that will be returned as results
+  out.file <- paste(in.prefix, format(as.Date(start_date), "%Y-%m-%d"),
+                    format(as.Date(end_date), "%Y-%m-%d"),
+                    "txt",
+                    sep = ".")
+  
+  out.file.full <- file.path(outfolder, out.file)
+  
+  # Results
+  results <- data.frame(file = out.file.full,
+                        host = PEcAn.remote::fqdn(),
+                        mimetype = "text/plain",
+                        formatname = "LDNDC_Climate",
+                        startdate = start_date,
+                        enddate = end_date,
+                        dbfile.name = out.file,
+                        stringsAsFactors = FALSE)
+  
+  
+  
+  if(length(nc_file) > 0){
+    print("At least one nc-file was found from the given path")
+  }
+  
+  
+  for(year in start_year:end_year){
+    
+    # Year
+    PEcAn.logger::logger.info(year)
+    
+    
+    old.file <- file.path(in.path, paste0(in.prefix, ".", year, ".nc"))
+    
+    
+    if(file.exists(old.file)){
+      
+      ## Open netCDF file
+      nc <- ncdf4::nc_open(old.file)
+      
+      # Data points are relational to this date
+      units <- nc$dim$time$units
+      
+      # Check that the simulation doesn't take place before there are data points
+      if(year == start_year & PEcAn.utils::datetime2cf(start_date, units, tz = "UTC") < 0){
+        PEcAn.logger::logger.severe("No data from the start date, consider postponing it")
+      }
+      
+      
+      # Convert the time fractions to be seconds by starting from the date in file's units
+      sec <- nc$dim$time$vals
+      sec <- udunits2::ud.convert(sec, unlist(strsplit(nc$dim$time$units, " "))[1], "seconds")
+      
+    
+      # Calculate the time steps
+      # 3600 * 24 = 86 400 (seconds in a day)
+      tstep <- 86400/(sec[2]- sec[1])
+      
+      
+      
+      
+      ## Determine the simulation days and calculate the start and end indexes
+      ## by using start_index and end_index functions. These functions are found
+      ## at the end of this file
+      
+      if(year != start_year & year != end_year){  
+        
+        # Simulation days
+        simu_days <- 1:PEcAn.utils::days_in_year(year)
+        
+        # Whole year, from first to last index
+        ind_s <- 1
+        ind_e <- which.max(sec)
+        
+      }else if(year == start_year & year != end_year){
+        
+        # Simulation days
+        simu_days <- seq(lubridate::yday(start_date),PEcAn.utils::days_in_year(year))
+        
+        # Check the function to see how the start index is calculated
+        ind_s <- start_index(units, start_date, sec)
+        ind_e <- which.max(sec)
+        
+      }else if(year != start_year & year == end_year){
+        
+        # Simulation days
+        simu_days <- seq(1:lubridate::yday(end_date))
```

Comment:
`seq` seems extra, you can also replace it all with `seq_len(lubridate::yday(end_date))`